### PR TITLE
Stats Date Picker: Add outside click handler to close popover

### PR DESCRIPTION
**Pull request by annacmc**
Related to #83141

## Proposed Changes

* This PR adds an outside click handler to close/dismiss popover when you click outside the element itself

## Testing Instructions

* Load the Calypso live branch
* Navigate to /stats/day/{site URL}
* Open the date picker using the following button:
	
image

* Check that if you click elsewhere on the page, the date picker popover is dismissed. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by a8ck3n** [comment]
Hey @annacmc. I see there was some discussion in Slack around this one. Just wanted to point out that the https://wpcalypso.wordpress.com/devdocs/design/info-popover already supports [this behaviour](https://github.com/Automattic/wp-calypso/blob/fe58883c4a2038651e25b8c1f9b620f77a5829f5/packages/components/src/popover/index.jsx#L150). It appears to be broken in our usage of the popover though.

I'll drop a note in the Slack convo with this as well. I wonder if we should investigate fixing that first before we consider adding our own handling on top.

File: client/components/stats-date-control/stats-date-control-picker.tsx [hunk 1]
```diff
@@ -29,8 +29,21 @@ const DateControlPicker = ( {
 		moment( dateRange.chartEnd ).format( 'YYYY-MM-DD' )
 	);
 	const infoReferenceElement = useRef( null );
+	const popoverRef = useRef( null );
 	const [ popoverOpened, togglePopoverOpened ] = useState( false );
 
+	useEffect( () => {
+		function handleClickOutside( event ) {
+			if ( popoverRef.current && ! popoverRef.current.contains( event.target ) ) {
+				togglePopoverOpened( false );
```
Comment (dognose24) [inline client/components/stats-date-control/stats-date-control-picker.tsx]: As the [PageModuleToggler](https://github.com/Automattic/wp-calypso/blob/911fd241a11ece3fb6619b206a8dca1d304b799b/client/blocks/stats-navigation/page-module-toggler.tsx#L74-L78) did previously, we can pass `togglePopoverOpened` as the `onClose` prop method to the `Popover` component for closing itself when clicking outside by the built-in functionality.

**Comment by annacmc** [comment]
> Hey @annacmc. I see there was some discussion in Slack around this one. Just wanted to point out that the https://wpcalypso.wordpress.com/devdocs/design/info-popover already supports [this behaviour](https://github.com/Automattic/wp-calypso/blob/fe58883c4a2038651e25b8c1f9b620f77a5829f5/packages/components/src/popover/index.jsx#L150). It appears to be broken in our usage of the popover though.
> 
> I'll drop a note in the Slack convo with this as well. I wonder if we should investigate fixing that first before we consider adding our own handling on top.

Thanks @a8ck3n for pointing this out, I'd totally missed it. I definitely think we should investigate fixing the existing supported behaviour if we can, before adding additional handling.

**Comment by annacmc** [comment]
Thanks @dognose24 for that info - it looks like passing `togglePopoverOpened` as the `onClose` prop method to the Popover component did the trick and seems to be working for me. 

I'd love some opinions from @grzegorz-cp @kangzj  and @a8ck3n  too because currently, it also closes when a shortcut is selected due to the nested elements.

**Review by kangzj — APPROVED** [review]
LGTM! Thanks for fixing this!

**Comment by a8ck3n** [comment]
Hi @annacmc ! This works well!

> I'd love some opinions from @grzegorz-cp @kangzj and @a8ck3n too because currently, it also closes when a shortcut is selected due to the nested elements.

That's the expected behaviour so I think we're good there. Per the final design update:

> Clicking a shortcut either by mouse or Enter key, will close the open state.

Ref: p1HpG7-p0i-p2

**Pull request merged**